.travis/.travis_nightly_build.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash -x
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+if [ -n "${DEPLOY_BUILD_OPTS}" ]; then
+  echo "Deploying to bintray"
+
+  BUILD_VERSION=$(grep -E "<revision>(.*)</revision>" pom.xml | cut -d'>' -f2 | cut -d'<' -f1)
+  echo "Current build version: $BUILD_VERSION${DEV_VERSION}"
+  mvn versions:set -DnewVersion="$BUILD_VERSION${DEV_VERSION}" -q -B
+  mvn versions:commit -q -B
```

Comment:
This will work. Thank you for coming up with this solution. Dealing with versions is very painful in Maven + CI/CD without unreliable -SNAPSHOTs.

Curious, where did you find inspiration on how achieve this version bumping?

Reply:
I tested it by uploading the generated jars without setting the version, while the bintray returns 409 conflicts when uploading the pom files. Then I feel that may be due the same checksum of the pom files that's already uploaded. Thus, bumping up version might be the option before publishing them. 
Another obstacle I found here is that travis doesn't work well with `mvn help:evaluate -Dexpression=revision -q -DforceStdout`, which is used to fetch properties from pom file; it returns nothing.
I've documented all the traps I met in that instruction.